src/typechecker/FStar.TypeChecker.TcTerm.fs

```diff
@@ -1913,6 +1941,7 @@ and tc_abs_check_binders env bs bs_expected  : Env.env                         (
           attrs@diff
         in
         let b = {binder_bv=hd;binder_qual=imp;binder_attrs=combine_attrs attrs attrs'} in
+        check_erasable_binder_attributes env b.binder_attrs t;
```

Comment:
checks that the erasable attribute only appears on binders that are really erasable